src/setup_payload/java/SetupPayloadParser-JNI.cpp

```diff
@@ -23,31 +36,53 @@ jint JNI_OnLoad(JavaVM * jvm, void * reserved)
 
 JNI_METHOD(jobject, fetchPayloadFromQrCode)(JNIEnv * env, jobject self, jstring qrCodeObj)
 {
-    CHIP_ERROR err;
+    CHIP_ERROR err                             = CHIP_NO_ERROR;
     const char * qrString                      = NULL;
     QRCodeSetupPayloadParser * qrPayloadParser = NULL;
     SetupPayload payload;
 
     qrString        = env->GetStringUTFChars(qrCodeObj, 0);
     qrPayloadParser = new QRCodeSetupPayloadParser(qrString);
+    env->ReleaseStringUTFChars(qrCodeObj, qrString);
```

Comment:
I think we've got `JniUtfString` helper for that.

Reply:
`JniUtfString` is in ChipDeviceController-JNI, which we shouldn't depend on here. #5738 separates it out, so it that merges first I will try to use it. Otherwise, will keep as is for now.